### PR TITLE
Warcraft3.info AKA support - new endpoint '/api/players/{battlenet}/aka'

### PR DESCRIPTION
**Pull request by Cepheid-UK**
Support for the Warcraft3.info Akas served at endpoint /api/players/{battlenet}/aka

snip1

Tested with Korean/Cyrillic script and works.

This will ping the Warcraft3.info endpoint once per hour to get all Akas, and cached them in memory.

### CODE REVIEW

File: W3ChampionsStatisticService/PlayerProfiles/PlayersController.cs [hunk 1]
```diff
@@ -128,6 +129,20 @@ public async Task<IActionResult> GetPlayerWinrate([FromRoute] string battleTag,
...
         }
+
+        [HttpGet("{battleTag}/aka")]
+        public IActionResult GetPlayerAka([FromRoute] string battleTag)
+        {
+            var player = _playerRepository.LoadAka(battleTag.ToLower());
+
+            return Ok(player);
+
+            // if (player != null) {
```
Comment (abeninski) [inline W3ChampionsStatisticService/PlayerProfiles/PlayersController.cs:140]: Remove commented code

File: W3ChampionsStatisticService/PlayerProfiles/PlayerRepository.cs [hunk 2]
```diff
@@ -145,6 +151,23 @@ public Task<PlayerOverview> LoadOverview(string battleTag)
             return null;
         }
 
+        public Player LoadAka(string battleTag) {
+
+            // string should be received all lower-case if the script it's written is permits lower case.
+            if (PlayerAkasCache == null) {
```
Comment (abeninski) [inline W3ChampionsStatisticService/PlayerProfiles/PlayerRepository.cs:157]: This is not needed

File: W3ChampionsStatisticService/PlayerProfiles/PlayerRepository.cs [hunk 3]
```diff
@@ -167,6 +190,28 @@ private Task<List<PlayerOverview>> LoadOverviews(int season)
...
+        private static async Task<List<PlayerAka>> FetchAkas() {
+
+            // list of all Akas
+            var war3infoApiKey = Environment.GetEnvironmentVariable("WAR3_INFO_API_KEY"); // CHANGE THIS TO SECRET FOR DEV
+            var war3infoApiUrl = "https://warcraft3.info/api/v1/aka/battle_net";
+
+            var httpClient = new HttpClient();
+            httpClient.DefaultRequestHeaders.Add("client-id", war3infoApiKey);
+
+            Console.WriteLine("Requesting Data from Warcraft3.Info aka API...");
```
Comment (abeninski) [inline W3ChampionsStatisticService/PlayerProfiles/PlayerRepository.cs:202]: Remove Console.Logs

**Comment by Cepheid-UK** [comment]
#14bb6ff includes these comments.

File: W3ChampionsStatisticService/PlayerProfiles/PlayerRepository.cs [hunk 4]
```diff
@@ -145,6 +151,20 @@ public Task<PlayerOverview> LoadOverview(string battleTag)
             return null;
         }
 
+        public Player LoadAka(string battleTag) {
+
```
Comment (abeninski) [inline W3ChampionsStatisticService/PlayerProfiles/PlayerRepository.cs:155]: remove empty line at start of method

File: W3ChampionsStatisticService/PlayerProfiles/PlayerRepository.cs [hunk 5]
```diff
@@ -167,6 +187,23 @@ private Task<List<PlayerOverview>> LoadOverviews(int season)
             return LoadAll<PlayerOverview>(t => t.Season == season);
         }
 
+        private static async Task<List<PlayerAka>> FetchAkas() {
+
```
Comment (abeninski) [inline W3ChampionsStatisticService/PlayerProfiles/PlayerRepository.cs:191]: remove empty line at start of method

File: W3ChampionsStatisticService/PlayerProfiles/War3InfoPlayerAkas/PlayerAka.cs [hunk 6]
```diff
@@ -0,0 +1,16 @@
+﻿namespace W3ChampionsStatisticService.PlayerProfiles.War3InfoPlayerAkas
+{
+    public class PlayerAka
+    {
+
```
Comment (abeninski) [inline W3ChampionsStatisticService/PlayerProfiles/War3InfoPlayerAkas/PlayerAka.cs:5]: remove empty line at start of method

**Review by abeninski — APPROVED** [review]

**Pull request merged**